lib/bin/run_test.js

```diff
@@ -63,6 +63,23 @@ var _ = require('lodash');
 var co = require('co');
 var exit = require('exit');  // process.exit that works on Windows
 var promiseUtil = require('../promise_util');
+var cleaned = false;
+
+function cleanup(exitCode) {
```

Comment:
A bit of a style issue perhaps, but I like wrapping variables used by only one function like

    var cleanup = (function() {
      var cleaned = false;
      return function(exitCode) {
        ..
      };
    })();

I find that helps when refactoring. It can be a bit hard to understand all invariants and cases where variables with module-wide scope are read/written.